File: Arbitration/MPArbitration/Model/BenchmarkDataItem.cs
using System.Text.Json.Serialization;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;

namespace MPArbitration.Model
{
    [Index("BenchmarkDatasetId","GeoZip","Modifiers","ProcedureCode",IsUnique =true)]
    public class BenchmarkDataItem
    {
        [JsonPropertyName("id")]
        public int Id { get; set; } = 0;

        [JsonPropertyName("benchmarks")]
        public string Benchmarks { get; set; } = string.Empty;  // JSON

        [JsonPropertyName("dataset")]
        public virtual BenchmarkDataset? Dataset { get; set; } = null;

        [JsonPropertyName("geoZip")]
        [StringLength(10)]
        public string GeoZip { get; set; } = string.Empty;

        [JsonPropertyName("modifiers")]
        [StringLength(255)]
        public string Modifiers { get; set; } = string.Empty;

        [JsonPropertyName("procedureCode")]
        [StringLength(20)]
        public string ProcedureCode { get; set; } = string.Empty;

        [JsonPropertyName("updatedBy")]
        [StringLength(60)]
        public string UpdatedBy { get; set; } = "";

        [JsonPropertyName("updatedOn")]
        public DateTime? UpdatedOn { get; set; } = null;

        [JsonPropertyName("benchmarkDatasetId")]
        public int BenchmarkDatasetId { get; set; }
    }
}

File: Arbitration/MPArbitration/ClientApp/src/app/model/benchmark-data-item.ts
export class BenchmarkDataItem {
    public id = 0;
    public benchmarks:string = '';
    public benchmarkDatasetId = 0;
    public geoZip = '';
    public modifiers = '';
    public procedureCode = '';
    public updatedBy = '';
    public updatedOn: Date|undefined;

    public get values():any {
        if(!this.benchmarks)
            return {};
        try {
            return JSON.parse(this.benchmarks);
        } catch {
            return {};
        }
    }

    constructor(obj?:any) {
        if(!obj)
            return;
        Object.assign(this, JSON.parse(JSON.stringify(obj)));
        if(this.updatedOn)
            this.updatedOn = new Date(obj.updatedOn);
    }
}
